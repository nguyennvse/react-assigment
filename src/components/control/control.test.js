import { render, cleanup, fireEvent } from "@testing-library/react"
import Controls from "./control";

it('should take a snapshot',()=>{
    const { asFragment } = render(<Controls/>)
    expect(asFragment(<Controls/>)).toMatchSnapshot();
});

it('should be control label',()=>{
    const {getByTestId} = render(<Controls label='Name'/>);
    expect(getByTestId('label')).toHaveTextContent('Name');
})

it('should be control label',()=>{
    const {getByTestId,container} = render(<Controls label='Name'/>);
    fireEvent.keyUp(getByTestId('control'))
    expect(container).toBeTruthy();
})
