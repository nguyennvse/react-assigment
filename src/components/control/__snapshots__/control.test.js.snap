// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full text-left px-3 mb-6"
  >
    <label
      class="block tracking-wide text-gray-700 text-xs font-bold mb-2"
      data-testid="label"
    />
    <input
      class="appearance-none block w-full text-gray-700 border rounded py-3 px-4 mb-3 leading-tight focus:outline-none focus:bg-white"
      data-testid="control"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
